ai/main.py

```diff
@@ -22,7 +23,11 @@ class QueryResponse(BaseModel):
 service_context = ServiceContext.from_defaults(embed_model=embed_model)
 set_global_service_context(service_context)
 
-storage_context = StorageContext.from_defaults(persist_dir="./storage")
+storage_context = StorageContext.from_defaults(
```

Comment:
Will this reload if the s3 data changes on a regular interval or would we need to restart the pod to get it to do that?

My general guess was we might need to have a thread that polls the bucket and reinitializes the storage context (or a sidecar that downloads it to a local file that can be read out)

Reply:
good point

Reply:
the quickest way to solve it is probably with a fastapi background task or sth, will look into it how this deals with global objects

Reply:
this would do it I think, probably not super nice doing it just with asyncio, there might be some fastapi features that are more suited, but it should work for the time being